libcxx/include/__new/allocate.h

```diff
@@ -29,7 +29,7 @@ _LIBCPP_CONSTEXPR inline _LIBCPP_HIDE_FROM_ABI bool __is_overaligned_for_new(siz
 #endif
 }
 
-template <class... _Args>
+template <class _Tp, class... _Args>
```

Comment:
Do you plan to actually use the type upstream? If not, I'd rather just pass the type to the `__libcpp_allocate` level. At that point the diff is already considerably smaller than before and we still gain a better API upstream.